assets/js/state/sagas/hosts.js

```diff
@@ -52,6 +52,7 @@ export function* deregisterHost({ payload }) {
   yield put(setHostDeregistering(payload));
   try {
     yield call(del, `/hosts/${payload.id}`);
+    payload.navigate?.('/hosts');
```

Comment:
```suggestion
    const { navigate } = payload;
    navigate('/hosts');
```

Reply:
But if `payload.navigate` doesn't exist and then we try to call `navigate()` we will get `Uncaught TypeError: navigate is not a function`.

With optional chaining if `navigate` doesn't exist, the call to `navigate()` just returns `undefined`, essentially a NOOP.

We need to do this because we only want to navigate back to `HostsList` from `HostDetails`. It doesn't really make sense to do it from `HostsList` because we are already there and it is updated by websocket events. So as a result [we don't pass in `navigate`](https://github.com/trento-project/web/blob/c52a178fe0df9c39910cf31ced48c4141427544d/assets/js/components/HostsList.jsx#L68) to the payload to make this a NOOP.